include/rocksdb/options.h

```diff
@@ -1063,6 +1073,21 @@ struct ReadOptions {
   // Default: false
   bool ignore_range_deletions;
 
+  // 3 params below are added for differential snapshots
+  // If true iterator will return internal keys instead of user keys.
+  // Default: false
+  bool internal_keys;
+
+  // Iterator start timestamp; only SST files with
+  // creation_time >= iter_start_ts  will be scanned.
+  // Default: 0 (include all files)
+  uint64_t iter_start_ts;
+
+  // Iterator will skip all internal keys if their seqnum
+  // is less than this param.
+  // Default: 0 (don't filter by seqnum)
+  SequenceNumber iter_start_seqnum;
```

Comment:
Wondering if both sequence number and timestamp are needed to solve your use case? Or would having one of them be enough?

Reply:
timestamp can be handled by the callback added to table_cache; this one is needed since for some cases like universal compaction the selectivity of timestamp-based filter isn't enough (in the extreme case 95% of the DB size is a single giant SST file, so we need to be able to slice and dice it to extract a small portion of it).